app/code/Magento/CacheInvalidate/Model/PurgeCache.php

```diff
@@ -131,10 +132,25 @@ private function sendPurgeRequestToServers($socketAdapter, $servers, $formattedT
                 $socketAdapter->read();
                 $socketAdapter->close();
             } catch (\Exception $e) {
-                $this->logger->critical($e->getMessage(), compact('server', 'formattedTagsChunk'));
-                return false;
+                $unresponsiveServerError[] = "Cache host: " . $server->getHost() . ":" . $server->getPort() .
+                    "resulted in error message: " . $e->getMessage();
+                continue;
             }
         }
+
+        if($errorCount = count($unresponsiveServerError) > 0) {
```

Comment:
Please move the assignment out of if condition and fix the styles (space after operators and commas) to pass the static tests